### ISSUE TITLE
`-Wincompatible-pointer-types` warnings in zig2.c when building from source

### ISSUE THREAD

**Issue by nektro** [ziglang/zig#15674]
### Zig Version

0.11.0-dev.3105+e46d7a369

### Steps to Reproduce and Observed Behavior

```
[nix-shell:~/src/zig/build]$ make install
-- Configuring zig version 0.11.0-dev.3105+e46d7a369
-- Configuring done
-- Generating done
-- Build files have been written to: /home/meghan/src/zig/build
Consolidate compiler generated dependencies of target zigcpp
[ 40%] Built target zigcpp
Consolidate compiler generated dependencies of target zig-wasm2c
[ 50%] Built target zig-wasm2c
Consolidate compiler generated dependencies of target zig1
[ 70%] Built target zig1
[ 75%] Running zig1.wasm to produce /home/meghan/src/zig/build/zig2.c
[ 80%] Running zig1.wasm to produce /home/meghan/src/zig/build/compiler_rt.c
Consolidate compiler generated dependencies of target zig2
[ 85%] Building C object CMakeFiles/zig2.dir/zig2.c.o
/home/meghan/src/zig/build/zig2.c: In function ‘objcopy_ElfFile_28true_29_emit__42204’:
/home/meghan/src/zig/build/zig2.c:346023:6: warning: assignment to ‘const uint8_t (*)[14]’ {aka ‘const unsigned char (*)[14]’} from incompatible pointer type ‘const uint8_t *’ {aka ‘const unsigned char *’} [-Wincompatible-pointer-types]
346023 |  t53 = (anon__42204_51){(uint8_t const *)((uint8_t const *)&objcopy_ElfFile_28true_29_parse__anon_42228__42228), (uintptr_t)14ul}.ptr;
       |      ^
/home/meghan/src/zig/build/zig2.c:346717:8: warning: assignment to ‘uint8_t (*)[4]’ {aka ‘unsigned char (*)[4]’} from incompatible pointer type ‘uint8_t *’ {aka ‘unsigned char *’} [-Wincompatible-pointer-types]
346717 |   t124 = t45.ptr;
       |        ^
/home/meghan/src/zig/build/zig2.c: In function ‘objcopy_ElfFile_28false_29_emit__42314’:
/home/meghan/src/zig/build/zig2.c:347923:6: warning: assignment to ‘const uint8_t (*)[14]’ {aka ‘const unsigned char (*)[14]’} from incompatible pointer type ‘const uint8_t *’ {aka ‘const unsigned char *’} [-Wincompatible-pointer-types]
347923 |  t53 = (anon__42314_51){(uint8_t const *)((uint8_t const *)&objcopy_ElfFile_28true_29_parse__anon_42228__42228), (uintptr_t)14ul}.ptr;
       |      ^
/home/meghan/src/zig/build/zig2.c:348626:8: warning: assignment to ‘uint8_t (*)[4]’ {aka ‘unsigned char (*)[4]’} from incompatible pointer type ‘uint8_t *’ {aka ‘unsigned char *’} [-Wincompatible-pointer-types]
348626 |   t123 = t45.ptr;
       |        ^
[ 90%] Building C object CMakeFiles/zig2.dir/compiler_rt.c.o
[ 95%] Linking CXX executable zig2
[ 95%] Built target zig2
[100%] Building stage3
[100%] Built target stage3
Install the project...
-- Install configuration: "Debug"
```

### Expected Behavior

build success with no warnings

**Comment by patterns** [comment]
I also noticed the incompatible warnings, but ignored them. If it is helpful, I started seeing them near the beginning of May. Memory fuzzy, but I thought it was after the switch to llvm16 (because I remember noticing the need for llvm16 at the time but not much else).

**Comment by patterns** [comment]
Just to update, the 3202 nightly is clean and resolved the incompatible warnings. I'm able to see the clean output both on FreeBSD-13.1 and 14.0

**Comment by nektro** [comment]
can confirm

**Comment by andrewrk** [comment]
Fix landed in 378264d404e71da878f9e6934c045ad64193877f.
